### ISSUE TITLE
ErrorMessageFactory cleaner approach?

### ISSUE THREAD

**Issue by spirosoik** [android10/Android-CleanArchitecture#71]
I was wondering if this can be done in an efficient way than the current one. Because if you have many type of exceptions then this will be a huge ``if/else if ``` to catch all the cases. 

I suppose that one solution is to separate the factories per feature but still I will duplicate the same code for application error or no internet connection to every factory.

**Comment by android10** [comment]
@spirosoik good point. For the size and scope of this example, that works but as your solutions scales things might get more difficult. What I would do is to implement an Abstract factory pattern that will provide factories which will create specific messages related with different features in your application (organized by package).
https://en.wikipedia.org/wiki/Abstract_factory_pattern

**Comment by spirosoik** [comment]
Yes something like that I was thinking. eg.

```java
public abstract class AbstractFactory {
   abstract String create(Context context, Exception exception); //or with implementation for standard errors
}

public class LoginErrorMessageFactory extends AbstractFactory {
	
   @Override
   public String create(Context context, Exception exception){
   
      return null;
   }
}```

**Comment by meierjan** [comment]
@android10 @spirosoik  shouldn't it be a abstract factory that creates factories per feature? You only did a abstract class that can be extended. I think there is a wrapping factory missing - isn't it?
